### ISSUE TITLE
Impl Error for SymmetricCypherError

### ISSUE THREAD

**Issue by Dr-Emann** [DaGenix/rust-crypto#352]
[crypto::symmetriccipher::SymmetricCipherError](http://ironframework.io/doc/crypto/symmetriccipher/enum.SymmetricCipherError.html) doesn't implement [std::error::Error](http://doc.rust-lang.org/stable/std/error/trait.Error.html). It probably should do so.

**Comment by kosta** [comment]
Yes, I agree. Would you accept a PR implementing std::error::Error?

**Comment by sseemayer** [comment]
I ran into the same issue, see PR #452
